aten/src/ATen/native/GroupedMMUtils.h

```diff
@@ -59,8 +59,9 @@ std::optional<c10::ScalarType> out_dtype
     }
   }
 
-  const auto out_dtype_ = out_dtype.value_or(kBFloat16);
-  TORCH_CHECK(out_dtype_ == kBFloat16, "Only bf16 high precision output types are supported for grouped gemm");
+  const auto out_dtype_ = out_dtype.value_or(mat_a.scalar_type());
```

Comment:
this is a behavior change for fp8 gemm? There it we don't specify dtype bfloat16 would be assumed, not mat_a scalar type